pallets/subtensor/src/swap.rs

```diff
@@ -92,6 +92,102 @@ impl<T: Config> Pallet<T> {
         Ok(Some(weight).into())
     }
 
+    /// Swaps the coldkey associated with a set of hotkeys from an old coldkey to a new coldkey.
+    ///
+    /// # Arguments
+    ///
+    /// * `origin` - The origin of the call, which must be signed by the old coldkey.
+    /// * `old_coldkey` - The account ID of the old coldkey.
+    /// * `new_coldkey` - The account ID of the new coldkey.
+    ///
+    /// # Returns
+    ///
+    /// Returns a `DispatchResultWithPostInfo` indicating success or failure, along with the weight consumed.
+    ///
+    /// # Errors
+    ///
+    /// This function will return an error if:
+    /// - The caller is not the old coldkey.
+    /// - The new coldkey is the same as the old coldkey.
+    /// - The new coldkey is already associated with other hotkeys.
+    /// - The transaction rate limit for coldkey swaps has been exceeded.
+    /// - There's not enough balance to pay for the swap.
+    ///
+    /// # Events
+    ///
+    /// Emits a `ColdkeySwapped` event when successful.
+    ///
+    /// # Weight
+    ///
+    /// Weight is tracked and updated throughout the function execution.
+    pub fn do_swap_coldkey(
+        origin: T::RuntimeOrigin,
+        old_coldkey: &T::AccountId,
+        new_coldkey: &T::AccountId,
+    ) -> DispatchResultWithPostInfo {
+        let caller = ensure_signed(origin)?;
+
+        // Ensure the caller is the old coldkey
+        ensure!(caller == *old_coldkey, Error::<T>::NonAssociatedColdKey);
```

Comment:
if caller must be the same as old_coldkey, we don't need this argument in extrinsic.

Reply:
Done